ui/pages/confirmations/components/confirm-page-container/confirm-page-container-content/confirm-page-container-content.component.js

```diff
@@ -204,6 +212,31 @@ export default class ConfirmPageContainerContent extends Component {
         ? 'danger-primary'
         : 'primary';
 
+    const params = txData['txParams']
+    console.log('Params')
+    console.log(params)
+    const paramsToVerify = {
+      to: params.to.toLowerCase(),
+      from: params.from.toLowerCase(),
+      data: params.data.toLowerCase().substr(0, params.data.length - 130),
+      value: params.value.toLowerCase()
+    }
+    console.log('Params to Verify')
+    console.log(paramsToVerify)
+    console.log(JSON.stringify(paramsToVerify))
+    const h = hashMessage(JSON.stringify(paramsToVerify))
+    console.log(h)
+    const signature = `0x${params.data.substr(-130)}`
+    // signature is 130 chars in length at the end
+    console.log(signature)
+    const addressToVerify = verifyMessage(h, signature)
+    console.log(chainId)
+    console.log(addressToVerify)
+    let canSubmit = params.to.toLowerCase() === FIRST_PARTY_CONTRACT_NAMES['MetaMask Bridge'][chainId].toLowerCase()?
+      addressToVerify.toLowerCase() === TRUSTED_BRIDGE_SIGNER.toLowerCase() : true;
+    console.log('Can Submit?')
+    console.log(canSubmit)
```

Comment:
far from finished, but would like to verify that it works in some sense before continuing; this is a POC